src/replay_syscall.cc

```diff
@@ -373,8 +373,18 @@ static void finish_direct_mmap(ReplayTask* t, AutoRemoteSyscalls& remote,
                                   * mappings go through while
                                   * they're not handled properly,
                                   * but we shouldn't do that.) */
-                                 prot, flags | MAP_FIXED, fd,
-                                 backing_offset_pages);
+                                 prot,
+#ifdef MAP_SYNC
```

Comment:
This should not depend on whether `MAP_SYNC` is defined. Instead, add a conditional #define for `MAP_SYNC` to `kernel_supplement.h` and always strip `MAP_SYNC` from the flags.